packages/core/src/storages/sitemap_request_list.ts

```diff
@@ -165,14 +234,28 @@ export class SitemapRequestList implements IRequestList {
         this.sitemapParsingProgress.pendingSitemapUrls = new Set(options.sitemapUrls);
     }
 
+    private isUrlMatchingPatterns(url: string): boolean {
+        return (
+            !this.urlExcludePatternObjects.some((patternObject) => {
+                const { regexp, glob } = patternObject;
+                return (regexp && url.match(regexp)) || (glob && minimatch(url, glob, { nocase: true }));
+            }) &&
+            (this.urlPatternObjects.length === 0 ||
+                this.urlPatternObjects.some((patternObject) => {
+                    const { regexp, glob } = patternObject;
+                    return (regexp && url.match(regexp)) || (glob && minimatch(url, glob, { nocase: true }));
+                }))
+        );
+    }
```

Comment:
I was thinking about something like this:

```suggestion
    private isUrlMatchingPatterns(url: string): boolean {
        const matchesSomeExcludePattern = this.urlExcludePatternObjects.some((patternObject) => {
                const { regexp, glob } = patternObject;
                return (regexp && url.match(regexp)) || (glob && minimatch(url, glob, { nocase: true }));
            })
        const matchesSomeIncludePattern = this.urlPatternObjects.some((patternObject) => {
                    const { regexp, glob } = patternObject;
                    return (regexp && url.match(regexp)) || (glob && minimatch(url, glob, { nocase: true }));
                })
        return (
            !matchesSomeExcludePattern &&
            (this.urlPatternObjects.length === 0 || matchesSomeIncludePattern)
        );
    }
```

Reply:
Oh, right, I guess I went a bit overboard... but I'd argue that my solution makes it ever so slightly even DRYier. 

Tomayto, tomahto, I think both of the solutions solve the same issue of readability in the condition, albeit each one a bit differently.

Reply:
Yeah, what you did is cool. Feel free to merge.